Danki2/Assets/Scripts/Stats/StatsManager.cs

```diff
@@ -1,27 +1,43 @@
 ﻿using System;
+using System.Collections.Generic;
 
 public class StatsManager
 {
     private readonly Stats _baseStats;
-    private readonly Stats _frameStats;
+    private readonly Dictionary<Stat, float> _statsCache;
+    private List<Func<Stat, float, float>> _pipes;
```

Comment:
I think it would be nicer to have the following interface:
```c#
public interface StatsPipe
{
    float ProcessStat(Stat stat, float value);
}
```
and change this to `List<StatsPipe> _pipes;`, instead of `List<Func<Stat, float, float>> _pipes;`, then `RegisterPipe` just asks for a `StatsPipe` and Effects tracker just implements this interface. I think it makes it clearer what the function is doing if you can give it a name (`ProcessStat`) rather than it just being a `Func<Stat, float, float>`, what do you think?

Reply:
I like that a lot